avocado/plugins/runner_nrunner.py

```diff
@@ -267,6 +267,15 @@ def run_suite(self, job, test_suite):
             job.interrupted_reason = f"Suite {test_suite.name} is disabled."
             return summary
 
+        spawner_name = test_suite.config.get("run.spawner")
+        spawner = SpawnerDispatcher(test_suite.config, job)[spawner_name].obj
+        if not spawner.is_operational():
+            msg = f'Spawner "{spawner_name}" is not operational, aborting execution of suite {test_suite.name}'
```

Comment:
My take is that each spawner is free to log as much as it wants about the failure.  The main goal of this is to stop having false positives and lead users into investigating the issue.

We can add some method of passing the failure reason from the spawner to the runner, and having the runner log it, but I feel it's unnecessary.

Reply:
Ok, I don't have a problem to leave this responsibility to spawners, but then I would add logs about failure to the PodmanSpawner. Because IMO `Spawner "PodmanSpawner" is not operational` doesn't give you many hints about have happened and what you need to fix.

Reply:
Indeed, if the spawner can detect it has a problem it should hopefully also be able to report what that problem is.

Reply:
For the podman spawner, I've added reporting for three possible errors:

1. The podman binary could not be found
2. The execution of the podman binary signaled failure (return code xxx)"
3. The podman binary did not report a suitable version (>= 3.0)